### PR TITLE
add model injection for plugins

### PR DESCRIPTION
**Pull request by MattMcFarland**
As discussed in IRC chat, please have a look!

### CODE REVIEW

**Comment by MattMcFarland** [comment]
Any chance this can go in?

File: index.js [hunk 1]
```diff
@@ -55,6 +55,11 @@ function dispatcher (hooks) {
     if (hooks.wrapInitialState) initialStateWraps.push(hooks.wrapInitialState)
     if (hooks.wrapReducers) reducerWraps.push(hooks.wrapReducers)
     if (hooks.wrapEffects) effectWraps.push(hooks.wrapEffects)
+    if (hooks.injectModels) {
+      hooks.injectModels.forEach(function (model) {
```
Comment (yoshuawuyts) [inline index.js]: I think you can just pass `setModel` in here without needing to pass the closure - shaves a stack in the stack trace :D

File: test.js [hunk 2]
```diff
@@ -29,6 +29,48 @@ tape('api: store.model()', (t) => {
...
+        {
+          namespace: 'namespaced',
+          state: {
+            'accessed': false
+          },
+          reducers: {
+            update: (state, data) => data
+          }
+        },
+        {
```
Comment (yoshuawuyts) [inline test.js]: can we put these on the line above? thanks! :grin:

File: test.js [hunk 3]
```diff
@@ -29,6 +29,48 @@ tape('api: store.model()', (t) => {
...
+      ]
+    })
+    const createSend = store.start()
+    const send = createSend('test', true)
+    send('namespaced:update', {
+      accessed: true
+    })
+    send('update', {
+      accessed: true
+    })
```
Comment (yoshuawuyts) [inline test.js]: can we make these oneliners? thank you :D

File: test.js [hunk 4]
```diff
@@ -29,6 +29,48 @@ tape('api: store.model()', (t) => {
...
+  t.test('should allow model injection', (t) => {
+    t.plan(1)
+    const store = barracks()
+    store.model({
+      state: {
+        accessed: false
+      }
+    })
+    store.use({
+      injectModels: [
```
Comment (yoshuawuyts) [inline test.js]: I think the "inject" word sounds scary; I never use it in day to day life. Maybs we could name it "addModels" instead?

**Pull request closed without merging**